prereise/gather/griddata/hifld/data_process/transmission.py

```diff
@@ -140,6 +140,15 @@ def map_lines_to_substations_using_coords(
         new_lines[e_sub] = substations.loc[primary_sub.values, "NAME"].values
         new_lines[f"{e_sub}_ID"] = primary_sub.values
 
+    # Remove lines which ended up assigned to the same substation after consolidation
+    if drop_zero_distance_line:
+        idx = new_lines["SUB_1_ID"].compare(new_lines["SUB_2_ID"]).index
+        print(
+            f"dropping {new_lines.shape[0] - len(idx)} lines having same endpoints "
+            "after substation consolidation"
+        )
+        new_lines = new_lines.loc[idx]
```

Comment:
Should we add this filter in `build_transmission` instead of here given `sub2line` approach might also have such issue or that's not the case?

Reply:
Do we ever plan to use the `sub2line` approach again? I think `line2sub` is better.

Reply:
Yes, maybe it won't hurt to add this filter there just in case of the potentials? Or should we delete `sub2line` approach to clean it up, but this is something we might do after the final feasibilities tuning right?

Reply:
I think it may already be included in `filter_lines_with_identical_substation_names`. within the `sub2line` block.

Reply:
I was thinking of the same thing so just to confirm with you on that:)